### DISCUSSION TITLE
parametrize variable name when passing in function it is giving error "fixture 'xxx' not found". Could you please help me..

### DISCUSSION THREAD

**Discussion by subhakswain** [pytest-dev/pytest#11529] [Q&A]
This is my conftest.py code----------

import pytest
from selenium import webdriver
from selenium.webdriver.common.desired_capabilities import DesiredCapabilities
from browserstack.local import Local
import os

def browserstack_local():
    global bs_local
    bs_local = Local()
    bs_local_args = { "key": "key", "forcelocal": "true" }
    bs_local.start(**bs_local_args)

@pytest.mark.parametrize("deviceconfiguration",[{ "deviceName": "Samsung Galaxy S22",
            "osVersion": "12.0"},{"deviceName": "iPhone 12",
            "osVersion": "14"}])
@pytest.fixture(scope="class")
def testdriver(deviceconfiguration):
    browserstack_local()
    # Define the remote WebDriver URL and desired capabilities
    BROWSERSTACK_URL = "https://username:key@hub-cloud.browserstack.com/wd/hub"  # Replace with your remote server URL
    desired_cap = {
        'bstack:options': {
            "deviceOrientation": "portrait",
            "realMobile": "true",
            "projectName": "Mobile Pytest Automation",
            "buildName": "Build #1",
            "sessionName": "Mobile Browser Automation",
            "local": "true"
            },
        "browserName": "Chrome",
        "browserVersion": "latest",
    }
    desired_cap.update(deviceconfiguration)
    driver = webdriver.Remote(
        BROWSERSTACK_URL,
        desired_cap
    )

    # Teardown - quit the driver after the test
    # def quit_driver():
    #     driver.quit()
    #
    # request.addfinalizer(quit_driver)

    return driver

And I am calling this function to another test class. And I am running that test class. Test class name is test_HomePage.py
And code is ---------

from Utilities import XLUtils
from helpers.config_helpers import get_base_url
from Pages.LoginPage import LoginPage
import os

class Test_HomePage():
    #baseURL = ReadConfig.getApplicationURL()
    baseURL=get_base_url()
    path = os.path.relpath(
        os.path.realpath(os.path.join(os.path.dirname(__file__), '..', 'testData', 'Automation_Data.xlsx')))

    def test_login_browser(self,testdriver):
        self.rows = XLUtils.getRowCount(self.path, 'username')
        lst_status = []
        for r in range(2, self.rows + 1):
            self.userName = XLUtils.readData(self.path, "username", r, 1)
            self.password = XLUtils.readData(self.path, "username", r, 2)
        self.driver=setthedriver
        print('base URL Is-----',self.baseURL)
        print('UserName',self.userName)
        print('Password',self.password )
        self.driver.get(self.baseURL)
        loginPage=LoginPage(self.driver)
        loginPage.click_login(self.userName,self.password)

I am running pytest -s -v test_HomePage.py
and getting error --------------
     **fixture 'deviceconfiguration' not found****
Please let me know why getting error deviceconfiguration not found which defined in parameterization. It will really helpful.

**Comment by The-Compiler** [comment]
Marks don't work on fixtures. See the [relevant section of the docs](https://docs.pytest.org/en/7.4.x/how-to/fixtures.html#parametrizing-fixtures) for how to do this properly.

Also, please [format code blocks](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks) properly in the future, your post is almost unreadable as-is.
